documentation/prplMesh-test-boardfarm.md

```diff
@@ -0,0 +1,80 @@
+# Introduction
+
+Current testing scheme is partially virtual.
+All test are done in so-called "dummy" mode.
+This mode operates without any interaction with WiFi device.
+Recently, the number of cases, when issue caused by changes wasn't detected by regular testing (using `test_flows.py`) but noticed during nightly testbed test, has increased.
+Yet, we don't have any means to provide automated testing on real hardware.
+So temporarily, all developers were forced to manually trigger testbed testing for each PR published as part of regular flow.
+CI has only one testbed set up available, so we can perform only single test at same moment.
+Also, testbed is able to perform only Cerification tests.
+We are interested in wider test scenarios, which cover project-specific bugs, corner cases and etc.
+
+Due to limitations of current testing approach, need of automated tests on real hardware raises.
+In order to speed up development process, it was decided by community to use [boardfarm](https://github.com/mattsm/boardfarm/) as a base for automated testing.
+This project can provide us with basic infrastructure for tests and save effort in terms of development and support.
+For details about integration process, design please see next sections.
+
+# Design
+
+By itself, [boardfarm](https://github.com/mattsm/boardfarm/) provides us with:
+1. Abstraction of test set-up
+1. Provides some basic classes of devices
+1. Some simple tests
+1. Handles connections to boards
+1. Provides test-suite organisation
+
+[boardfarm](https://github.com/mattsm/boardfarm/) already has some representation of some general devices (like Linux, OpenWRT device).
+But our test flows involve more components and require more info about devices\[1\].
+So we have to expand base classes with representation of `prplWRT` devices capable handling Agent, Controller and Sniffer.
```

Comment:
This sentence is unclear. I would expect to expand the base classes with EasyMesh device, which can be an agent or controller, so:
```
We expand base classes with the representation of `EasyMesh` devices capable of running as Controller, Agent, or both.
```
What does the sniffer has to do with it?

Reply:
Sniffer is not EasyMesh entity, it's standalone docker image/device in setup. Order of words has confused.
Moved Sniffer mention before the mention of EasyMesh-specific entities, to avoid confusion that Sniffer has anything to do with EasyMesh specific entities.